### ISSUE TITLE
Still working with new platform driver?

### ISSUE THREAD

**Issue by JeGr** [bernmc/flashstor-trueNAS-fancontrol#3]
Hi,

as I'm currently looking into buying a Flashstor and running TrueNAS on it, I found your project but also found a comment with that link at reddit:

https://www.reddit.com/r/asustor/comments/16xp8xt/comment/kjdzwsz/

about the platform-driver being updated and the compiling no longer working cleanly. Or is it a just a naming thing from renaming it87 to asustor-it87 now?

Just wanted to check the possibility before jumping in the compiling happyness :)

Thanks,
Jens

**Comment by bernmc** [comment]
Yes, Mafredi has updated his platform driver, so the it87 branch no longer exists - it's all in one. I will review and redo the documentation when I get a chance, but it appears that it'll need us to explicitly load the asustor-it87 module, and or explicitly prevent loading the it87 module (the two conflict). Probably not a big deal - just a few extra steps.

**Comment by JeGr** [comment]
Just wanted to add:

Recently bought a Flashstor 12 and tried it. As I overread your line

`git checkout it87`

I didn't checkout the it87 branch and just continued with

`make` and `make install`. That didn't do it though - obviously. But I re-checked the `Makefile` and found it containing a `make dkms` phrase, so tried autobuilding that via the DKMS toolset and actually it worked like a charm. After `apt install dkms` and `make dkms` the fan immediatly spun up as it built the module and autoloaded it afterwards. So I did

```
# skipped that:
# > git checkout it87

make
make dkms

# skipped depmod
# skipped modprobe
```
and it worked fine. Rebooted to check, checked with `lsmod` and module was loaded, sensors showing up.

I then grabbed your scripts - big thanks for them! - and tried running it -> also worked like charm. Changed minimal RPM to 80 so you REALLY hear the fan, rebooted again, fan was off then spun on minimal and after system booted you immediatly heared it spin up to 80% and checking via SSH the script was running fine.

Just wanted to report back to perhaps help you adapt the README or give a hint for an alternative with DKMS.

One last thing: would it be possible to include some LED settings in your monitoring script like setting the blinking status LED to "always on" after the script starts (as then the system should be up anyways) and perhaps (just for the fun of it) use colors of one of the other LEDs to show the fan status or speed?

Big cheers to you!

Thanks,
Jens

**Comment by phgarman** [comment]
Hmm, Tried to follow your instructions @JeGr  and not working for me.  Wondering what I'm doing wrong!  Any ideas on where to go troubleshooting?

**Comment by bernmc** [comment]
I haven't had a chance to try Jens' solution yet but will get around to it.
It would be useful if peops quoted the version of TrueNAS they were using when commenting so we can keep of track of what works where.

**Comment by JeGr** [comment]
> Hmm, Tried to follow your instructions @JeGr and not working for me. Wondering what I'm doing wrong! Any ideas on where to go troubleshooting?

I had to do it once again after the last stable update (23.10 cobia latest patchlevel). So went through it once again and it worked.

Did you follow 1) and 1.5) at first? Also 2) should not show sensors for the fan (it didn't for me after the update). So on to 3).
Checked out the asustor-platform-driver and "cd'ed" into it. Skipped the it87 checkout. Then did an `apt install dkms` and then went with `make` and `make dkms` again.
After completing I did a insmod/modprobe of the module and afterwards checked out the sensors - now showing the fan. Then followed the steps about the scripts and temp_monitor to get the fans "up to speed" so to speak ;)

Cheers

**Comment by sb10** [comment]
I can confirm that with TrueNAS Scale 24.04.2.3 this is all you need:
```
sudo su
install-dev-tools
sensors
[no info]
git clone https://github.com/mafredri/asustor-platform-driver
cd asustor-platform-driver
apt install dkms
make
make dkms
cd ..
sensors
[fan info]
```

Then you skip other steps and just go straight to creating the sh scripts (modified to also only do the above).

**Comment by bernmc** [comment]
Thanks @sb10 and @JeGr . I checked this again today and confirm it works with **TrueNAS Scale Dragonfish (24.04)**. I'll update the instructions.

I also tried an install on the latest **TrueNAS Scale Electric Eel Release Candidate (24.10-RC.2**) and it failed at the 'make' stage. I'm not going to investigate this further for now as things will change until the full release version. 

So for now (21 Oct 2024):

- Confirmed **working** for TrueNAS 24.04 (Dragonfish)
- Confirmed **NOT working** for TrueNAS 24.1 RC (Electric Eel)

The original check script will also no longer work (used modprobe etc), so I will re-write that.
